package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,509 @@
+
+    :Returns:
+       :class:`numpy.ndarray` containing the transformed coordinates of *mobile* 
+    """
+    lmob = mobile - mobile.mean(axis=0)
+    targ_com = reference.mean(axis=0)
+    ltarg = reference - targ_com
+    R, rms = rotation_matrix(ltarg, lmob, weights=weights)
+    fitted = np.dot(lmob, np.array(R))
+    fitted += targ_com
+    return fitted
+
+
+def rmsd(mobile, reference, weights=None):
+    """
+    Utility function to calculate the rmsd between the coordinates in 
+    the [n, 3] numpy array *mobile* and those in the [n, 3] array
+    *reference*, with optional mass weighting.
+
+    :Arguments:
+      *mobile*
+        :class:`numpy.ndarray` containing the coordinates to be fitted
+
+      *reference*
+        :class:`numpy.ndarray` containing the target coordinates
+
+      *weights*
+        :class: numpy.ndarray`  (optional) for mass-weighted fitting
+
+    :Returns:
+       float, the RMS deviation between *mobile* and *reference* after
+       least-squares fiting (with optional mass-weighting) 
+    """
+    lmob = mobile - mobile.mean(axis=0)
+    targ_com = reference.mean(axis=0)
+    ltarg = reference - targ_com
+    R, rms = rotation_matrix(ltarg, lmob, weights=weights)
+    return rms
+
+
+class Fasu(object):
+    def __init__(self,
+                 universe,
+                 atom_selection='name *',
+                 start=None,
+                 stop=None,
+                 step=None):
+        """
+        A "Fasu" is a convenience wrapper round an MDAnalysis universe,
+        providing slicing and pseudo-direct access methods for any 
+        trajectory format. In fact Fasu stands for "Filtered and sliced universe"
+        (filtered according to atom selection and sliced according to start/stop/step values).
+
+        :Arguments:
+          *universe*
+            :class:`MDAnalysis.Universe` object
+
+          *atom_selection* 
+            a valid selection string for 
+            :meth:`MDAnalysis.core.AtomGroup.AtomGroup.select_atoms`
+
+          *start*
+            int, the index (zero-based) of the first frame in the trajectory
+            to include in the PC analysis
+
+          *stop*
+            int, the index (zero-based) of the last frame in the trajectory 
+            to include in the PC analysis
+
+          *step*
+            int, the increment between frames to be included in the analysis
+        """
+        self._u = universe
+        self._sel = self._u.select_atoms(atom_selection)
+        self.natoms = len(self._sel)
+        if start is None:
+            self._start = 0
+        else:
+            self._start = start
+        if stop is None:
+            self._stop = self._u.trajectory.n_frames
+        else:
+            self._stop = stop
+        if step is None:
+            self._step = 1
+        else:
+            self._step = step
+
+        self.numframes = abs(self._stop - self._start) / self._step
```

Comment:
This needs to be an integer, but in Python 3 it will be a floating point number. Please make sure `self.numframes` is guaranteed to be an integer with `int()` or by using `//` for integer division.